File: toolkit/ltx_only.py
from toolkit.throughput_profiles import is_ltx_only_mode_enabled


def is_ltx_training_process(process_cfg: dict) -> bool:
    if not isinstance(process_cfg, dict):
        return True
    model_cfg = process_cfg.get("model", {})
    if not isinstance(model_cfg, dict):
        return True
    arch = str(model_cfg.get("arch", "") or "").strip().lower()
    name_or_path = str(model_cfg.get("name_or_path", "") or "").strip().lower()
    if arch.startswith("ltx2"):
        return True
    if "lightricks/ltx-2.3" in name_or_path or "ltx-2.3" in name_or_path:
        return True
    return False


def validate_ltx_only_config(config: dict):
    if not is_ltx_only_mode_enabled():
        return
    process_list = config.get("config", {}).get("process", [])
    for idx, process_cfg in enumerate(process_list):
        process_type = str(process_cfg.get("type", "")).strip().lower()
        if process_type not in {"sd_trainer", "diffusion_trainer", "ui_trainer"}:
            continue
        if not is_ltx_training_process(process_cfg):
            raise ValueError(
                "LTX-only mode is enabled. Non-LTX training jobs are blocked. "
                f"Found non-LTX process at config.process[{idx}] (type={process_type}). "
                "Set AITK_ALLOW_NON_LTX=1 to override."
            )

File: ui/src/server/ltxOnly.ts
export const parseEnvBool = (value: string | undefined | null, defaultValue: boolean): boolean => {
  if (value === undefined || value === null) return defaultValue;
  const normalized = String(value).trim().toLowerCase();
  if (['1', 'true', 'yes', 'on'].includes(normalized)) return true;
  if (['0', 'false', 'no', 'off'].includes(normalized)) return false;
  return defaultValue;
};

export const isLtxOnlyMode = (): boolean => {
  if (parseEnvBool(process.env.AITK_ALLOW_NON_LTX, false)) return false;
  return parseEnvBool(process.env.AITK_LTX_ONLY_MODE, true);
};

const isLtxModel = (model: any): boolean => {
  const arch = String(model?.arch ?? '').toLowerCase().trim();
  const nameOrPath = String(model?.name_or_path ?? '').toLowerCase().trim();
  return arch.startsWith('ltx2') || nameOrPath.includes('lightricks/ltx-2.3') || nameOrPath.includes('ltx-2.3');
};

export const isLtxJobConfig = (jobConfig: any): boolean => {
  const processList = jobConfig?.config?.process;
  if (!Array.isArray(processList)) return true;
  for (const processCfg of processList) {
    const type = String(processCfg?.type ?? '').toLowerCase().trim();
    if (!['sd_trainer', 'diffusion_trainer', 'ui_trainer'].includes(type)) {
      continue;
    }
    if (!isLtxModel(processCfg?.model)) {
      return false;
    }
  }
  return true;
};
